Ivan_Boichuk_1_HW.cpp

```diff
@@ -0,0 +1,77 @@
+
+
+#include <cmath>
+#include <iostream>
+
+int main() {
+
+  constexpr int errorExitCode{-1};
+  constexpr int successExitCode{};
+
+  int lowRangeA{2};
+  int highRangeA{255};
+  int a{};
+
+  constexpr auto incorrectInputErrorText{
+      "You entered an incorrect number, try entering the number again"};
+
+  std::cout << "Please enter number in range [" << lowRangeA << ".."
+            << highRangeA << "] for a\n";
+  std::cin >> a;
+
+  if (a < lowRangeA or a > highRangeA) {
+    std::cout << incorrectInputErrorText;
+    return errorExitCode;
+  }
+
+  int lowRangeB{2};
+  int highRangeB{255};
+  int b{};
+
+  std::cout << "Please enter number in range [" << lowRangeB << ".."
+            << highRangeB << "] for b\n";
+  std::cin >> b;
+
+  if (b < lowRangeB or b > highRangeB) {
+    std::cout << incorrectInputErrorText;
+    return errorExitCode;
+  }
+
+  int lowRangeC{2};
+  int highRangeC{255};
+  int c{};
+
+  std::cout << "Please enter number in range [" << lowRangeB << ".."
+            << highRangeB << "] for c\n";
+  std::cin >> c;
+
+  if (c < lowRangeC or c > highRangeC) {
+    std::cout << incorrectInputErrorText;
+    return errorExitCode;
+  }
+
+  long long D{static_cast<long long>(b * b) -
+              (static_cast<long long>(4) * a * c)};
```

Comment:
в вашем выражении достаточно написать превести один из аргументов к long long для того чтобы все выражения было как long long.
```suggestion
  long long D{(b * b) -
              (4LL * a * c)};
```
т.е. можно использовать суфиксы, в даном случае суфикс LL говорит что литерал по умолчанию будет не типа int а типа long long

Reply:
Тут студия нервничала из за возможного переполнения при умножении "b * b". Как поведет себя программа? Результат умножения (b * b) поместит в переменную типа long long  а потом вычтет из нее результат (4LL * a * c)?